src/useQueuedForceUpdate.ts

```diff
@@ -0,0 +1,41 @@
+import React from "react"
+
+export type ForceUpdate = () => void
+
+let insideRender: boolean = false
+let forceUpdateQueue: ForceUpdate[] = []
+
+export function useQueuedForceUpdate(forceUpdate: ForceUpdate): ForceUpdate {
+    return () => {
+        if (insideRender) {
```

Comment:
If the component invalidates itself, we don't have to schedule anything since the component si already being rendered with the late-most values. Can you add a check for that?

I was hoping we could do something similar for all components, but now I am not sure. The idea is that when component starts to rendering (before the `useLayoutEffect` has a chance to run) it would remove itself from `forceUpdateQueue`.

Reply:
Or rather instead of *removing itself from queue* (as it would require some lookup), there would be a check in this else branch like 
```else if (!wasAlreadyRendered) {
  forceUpdate()
}
````

Reply:
I think this might be possible, but I'm not sure it wouldn't have any unwanted side-effects so I'd rather leave it as-is.
As this should also not happen that often and only affects a single component, I don't think this would matter much performance wise.